pkg/channel/fanout/fanout_event_handler.go

```diff
@@ -219,6 +218,7 @@ func createEventReceiverFunction(f *FanoutEventHandler) func(context.Context, ch
 			reportArgs := channel.ReportArgs{}
 			reportArgs.EventType = evnt.Type()
 			reportArgs.Ns = ref.Namespace
+			reportArgs.EventScheme = "http"
```

Comment:
Is there a reason why you are always using http here? This isn't going to work correctly, as the fanout event handler may send to an https destination

Reply:
hmm , the way its there in code for me it seem to take the scheme for first subscriber . would that all subscribers would have scheme ?

Reply:
This one is complicated because the subscribers could have different schemes and we are reporting a metric based on the total time to dispatch the event (across potentially multiple schemes). My initial impression would be to check which schemes the subscribers are and:
1. All http -> report metric with http tag
2. All https -> report metric with https tag
3. Mix of http and https -> report metric twice, once with http and once with https tag

@pierDipi thoughts on this?

Reply:
seems logical to me . updated the change